### ISSUE TITLE
A testing bug in FCC's React Simple Counter Challenge

### ISSUE THREAD

**Issue by Banhawy** [freeCodeCamp/freeCodeCamp#17545]
I found a subtle bug while doing the “React: Write a Simple Counter” challenge.

If you look at the code below you’ll see that I binded *this* in *this.reset* to the *this .decrement method.*
It should be:
```jsx
this.reset = this.reset.bind(this);
```

The consequence of this error is that clicking the reset button which calls this.reset does not reset the counter and decrements it instead.

However, when I run the tests, all the tests are passed and the error is not detected!

I think we need to add an additional test for that specific scenario.

#### Link to the page with the problem:
[React Simple Counter](https://learn.freecodecamp.org/front-end-libraries/react/write-a-simple-counter)

#### My buggy code
```jsx


class Counter extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      count: 0
    };
    // change code below this line
    this.increment = this.increment.bind(this);
    this.decrement = this.decrement.bind(this);
    this.reset = this.decrement.bind(this);  //Wrong binding
    // change code above this line
  }
  // change code below this line
  increment(){
    const count = this.state.count + 1;
    this.setState({count});
  }
  decrement(){
    const count = this.state.count - 1;
    this.setState({count});
  }
  reset(){
    const count = 0;
    this.setState({count});
  }
  // change code above this line
  render() {
    return (
      <div>
        <button className='inc' onClick={this.increment}>Increment!</button>
        <button className='dec' onClick={this.decrement}>Decrement!</button>
        <button className='reset' onClick={this.reset}>Reset</button>
        <h1>Current Count: {this.state.count}</h1>
      </div>
    );
  }
};
```
#### browser and operating system

* Browser Name: Chrome 
* Browser Version: 67
* Operating System: Mac OS X 10_13_3

**Comment by KBPsystem777** [comment]
I run to this same error just now. Accidentally, I entered `this.reset = this.decrement.bind(this)` which is wrong but it passed the test.

**Comment by thecodingaviator** [comment]
I think we need to check this with a RegEx. What are your opinions @ezioda004 and @erictleung ?

**Comment by chaomt** [comment]
Is there any update on this issue? I could work on this bug.

**Comment by thecodingaviator** [comment]
@chaomt As of yet, we have not come to a conclusion about what to do on this issue. Although, since this is a confirmed issue, please suggest how you'd like to go ahead with this over here

**Comment by chaomt** [comment]
@thecodingaviator 
In terms of the buggy code provided in this issue, I think only the last test case is wrong. 

- Left part is the original version
- Right part is the new version I made
```I added one more "click", and changed the values for "secondValue" and "thirdValue".```
![Screen Shot 2019-03-30 at 3 03 15 AM](https://user-images.githubusercontent.com/42958064/55273083-91712d00-529c-11e9-935c-f10b611e2984.png)

I checked if my changes work on local host.

After I made the change, we can see that the last test fails when I provide the wrong solution codes.
![After Change - Wrong Codes](https://user-images.githubusercontent.com/42958064/55273097-b5347300-529c-11e9-8a90-66c32d4cb9c9.png)

And the last test passes when I provide the correct solution codes.
![After Change - Correct Codes](https://user-images.githubusercontent.com/42958064/55273102-c2e9f880-529c-11e9-8b03-04b02d54c3d8.png)
